core/autocrypt/crypto.py

```diff
@@ -0,0 +1,574 @@
+# -*- coding: utf-8 -*-
+# vim:ts=4:sw=4:expandtab
+
+"""Crypto implements GPG operations needed for Autocrypt.
+API is copied from bingpg.py
+"""
+
+from __future__ import print_function, unicode_literals
+import logging
+import os
+import glob
+import sys
+from base64 import b64encode
+import re
+import getpass
+from operator import attrgetter
+from pgpy import PGPKey, PGPUID, PGPMessage, PGPKeyring
+from pgpy.constants import PubKeyAlgorithm, KeyFlags, HashAlgorithm
+from pgpy.constants import SymmetricKeyAlgorithm, CompressionAlgorithm
+
+logger = logging.getLogger(__name__)
+
+
+def key_base64(pgpykey):
+    """Base 64 representation of key bytes.
+
+    :param key: key (either public or private)
+    :type key: PGPKey
+    :return: Base 64 representation of pgpykey bytes
+    :rtype: string
+
+    """
+    assert type(pgpykey) == PGPKey
+    keybytes = key_bytes(pgpykey)
+    keybase64 = b64encode(keybytes)
+    return keybase64
+
+
+def key_bytes(pgpykey):
+    """Key bytes.
+
+    :param key: key (either public or private)
+    :type key: PGPKey
+    :return: key bytes
+    :rtype: string
+
+    """
+    assert type(pgpykey) == PGPKey
+    if sys.version_info >= (3, 0):
+        keybytes = bytes(pgpykey)
+    else:
+        keybytes = pgpykey.__bytes__()
+    return keybytes
+
+
+def cached_property(f):
+    # returns a property definition which lazily computes and
+    # caches the result of calling f.  The property also allows
+    # setting the value (before or after read).
+    def get(self):
+        propcache = self.__dict__.setdefault("_property_cache", {})
+        key = f.__name__
+        try:
+            return propcache[key]
+        except KeyError:
+            x = self._property_cache[key] = f(self)
+            return x
+
+    def set(self, val):
+        propcache = self.__dict__.setdefault("_property_cache", {})
+        propcache[f.__name__] = val
+    return property(get, set)
+
+
+class Crypto(object):
+    """ GPG operations for Autocrypt using PGPy. """
+    # TODO: Write in identity YAML file the keys, so it won't be needed
+    # to import/export in different files.
```

Comment:
i don't understand this comment ... what do you refer to when you say "identity YAML file"?